### PR TITLE
add support for more fujifilm tags

### PR DESCRIPTION
**Pull request by VJSchneid**
I made some tests with the Fujifilm X-T4 and added/improved some missing makernote tags.

The following Fujifilm tags where added:
 - ColorTemperature
 - WhiteBalanceFineTune
 - HighIsoNoiseReduction
 - Clarity
 - FocusArea
 - FocusPoint
 - LensModulationOptimizer
 - GrainEffectRoughness
 - GrainEffectSize
 - ColorChromeEffect
 - ColorChromeFXBlue
 - ImageNumber

Additionally the print values for the following Fujifilm tags where extended:
 - Sharpness
 - WhiteBalance
 - Color
 - FlashMode
 - Shadow/Highlight Tone
 - Film Mode

### CODE REVIEW

File: src/fujimn_int.cpp [hunk 1]
```diff
@@ -53,37 +53,59 @@ namespace Exiv2 {
...
         {  769, N_("Fluorescent (warm white)") },
         {  770, N_("Fluorescent (cool white)") },
         { 1024, N_("Incandescent")             },
+        { 1536, N_("Underwater")               },
         { 3480, N_("Custom")                   },
+        { 3840, N_("Custom 1")                 },
+        { 3841, N_("Custom 2")                 },
+        { 3842, N_("Custom 3")                 },
+        { 4080, N_("Kelvin")                   },
         { 3480, N_("Custom")                   } // To silence compiler warning
```
Comment (kmilos) [inline src/fujimn_int.cpp:84]:
```suggestion
        { 4080, N_("Kelvin")                   }
```

File: src/fujimn_int.cpp [hunk 2]
```diff
@@ -53,37 +53,59 @@ namespace Exiv2 {
...
-        {   0, N_("Normal")                 },
-        { 256, N_("High")                   },
-        { 512, N_("Low")                    },
-        { 768, N_("Monochrome")             },
-        { 769, N_("Monochrome + R Filter")  },
-        { 770, N_("Monochrome + Ye Filter") },
-        { 771, N_("Monochrome + G Filter")  },
-        { 784, N_("Sepia")                  },
-        { 768, N_("Monochrome")             } // To silence compiler warning
+        {    0, N_("Normal/0")               },
```
Comment (kmilos) [inline src/fujimn_int.cpp:89]: Same as above, is there a more consistent way to present the value+label combo?

File: src/fujimn_int.cpp [hunk 3]
```diff
@@ -138,13 +211,35 @@ namespace Exiv2 {
...
+        { -40, N_("+2.5") },
+        { -32, N_("+2")   },
+        { -24, N_("+1.5") },
+        { -16, N_("+1")   },
+        {  -8, N_("+0.5") },
+        {   0, N_("0")    },
+        {   8, N_("-0.5") },
+        {  16, N_("-1")   },
+        {  24, N_("-1.5") },
+        {  32, N_("-2")   },
```
Comment (kmilos) [inline src/fujimn_int.cpp:226]:
```suggestion
        {  32, N_("-2")   }
```

File: src/fujimn_int.cpp [hunk 4]
```diff
@@ -53,37 +53,59 @@ namespace Exiv2 {
 
     //! Sharpness, tag 0x1001
     constexpr TagDetails fujiSharpness[] = {
-        { 1, N_("Soft mode 1") },
-        { 2, N_("Soft mode 2") },
-        { 3, N_("Normal")      },
-        { 4, N_("Hard mode 1") },
-        { 5, N_("Hard mode 2") }
+        {   0, N_("-4")             },
+        {   1, N_("Soft mode 1/-3") },
```
[the remark this answers is not in the document]
Comment (VJSchneid) [reply src/fujimn_int.cpp:57]: Good point. The X-T4 (and probably all other recent Fujifilm cameras) display the sharpness as number (-4,-3,-2,-1,0,+1,+2,+3,+4). I think older cameras display this setting as text and support fewer options.

**Comment by postscript-dev** [comment]
Thank you for taking time to contribute this.

Exiv2 collaborates with another open source project called [ExifTool](https://exiftool.org/). Their website contains a more complete list of tags and values that have been discovered for the different camera types. ExifTool's FujiFilm tags can be found [here](https://exiftool.org/TagNames/FujiFilm.html) and the source code (containing all the conditions, extra info and pretty printing) [here](https://github.com/exiftool/exiftool/blob/master/lib/Image/ExifTool/FujiFilm.pm). There is some additional information on the websites, that you may consider including (e.g., extra entries in the arrays).

File: src/fujimn_int.cpp [hunk 5]
```diff
@@ -53,15 +53,15 @@ namespace Exiv2 {
...
-        {   2, N_("Soft mode 2/-2") },
-        {   3, N_("Normal/0")       },
-        {   4, N_("Hard mode 1/+2") },
-        {   5, N_("Hard mode 2/+3") },
-        {   6, N_("+4")             },
-        { 130, N_("-1")             },
-        { 132, N_("+1")             }
+        {   0, N_("-4")               },
+        {   1, N_("-3 (Soft mode 1)") },
+        {   2, N_("-2 (Soft mode 2)") },
```
Comment (kmilos) [inline src/fujimn_int.cpp:58]: This is a bit controversial (not symmetrical): soft 1 = -3 but hard 1 = +2; I would expect soft/hard 1 = -/+2 and soft/hard 2 = -/+3... any way to confirm these?

**Comment by piponazo** [comment]
Hi everybody. Thanks @VJSchneid for the contribution. I am just taking a look to some open PRs and I was wondering if we could give the final push to this PR to get it merged. I am not familiar with the Fujifilm part of the code, but it looks like you added support for many new tags. It would be great to incorporate these changes into Exiv2.

Could you please rebase your branch in top of `main` so that the CI pipeline runs again? It looks like some of the CI jobs are in red because some hiccup in Github actions. Maybe we get those CI checks into "green" state just by rebasing your branch.

**Comment by piponazo** [comment]
Hi @VJSchneid again, and sorry for taking so long to come back to you. After you rebased the branch, most of the CI jobs failed. There has been lot of activity in the last month in `main` and there seems to be some conflicts now. Do you need any help with this? I just would not like to see this contribution getting stuck for not giving it enough attention (from our side).

**Comment by VJSchneid** [comment]
Hi @piponazo, sorry that this response took so long. I rebased my branch again and hopefully everything is in sync now.
Additionally some missing testfiles were also updated. Unfortunately I get some errors when runnung ``ctest`` with ``Exif.CanonCs.LensType`` but this should be independent of these changes. The fujifilm stuff seems to pass :)

**Comment by VJSchneid** [comment]
Hi @piponazo, I have merged the latest changes with this PR again. Is there anything more I can do to speed things up?

**Comment by piponazo** [comment]
> Hi @piponazo, I have merged the latest changes with this PR again. Is there anything more I can do to speed things up?

Sorry for my late reply, I am not participating much in the project lately.

Everything looks green, let's merge it!

**Review by piponazo — APPROVED** [review]

**Pull request merged**